app/controllers/hosts_controller.rb

```diff
@@ -58,8 +58,11 @@ def index(title = nil)
         @hostgroup_authorizer = Authorizer.new(User.current, :collection => @hosts.map(&:hostgroup_id).compact.uniq)
         render :index if title && (@title = title)
       end
-      format.yaml { render :text => search.all(:select => "hosts.name").map(&:name).to_yaml }
-      format.json
+      format.csv do
+        @hosts = search.includes(included_associations - [:host_statuses, :token, :compute_resource])
+        send_data @hosts.to_csv(columns: [:id, :name, :os, :environment, :model, :hostgroup, :last_report]),
+                  filename: "hosts-#{Date.today}.csv"
```

Comment:
One note, this will easily eat all memory of a server process for big tables like reports or audits. While `send_data` streams (which is exactly what we want to do), method `to_csv` creates in-memory string. We already treat memory badly in Foreman and we do not want to make it worse, so my point is, can you rewrite this into a helper method that will stream the data row by row. Ideally you can read the rows in batches using `find_in_batches` so it's also "streaming" from DB directly.